### ISSUE TITLE
Feasibility question: Implementation of additional expression languages

### ISSUE THREAD

**Issue by cdjones32** [neuland/jade4j#127]
Hello, 

TLDR; The main question: would the Jade4J project consider taking pull requests for a feature that allowed the expression language to be specified? 

Background

I am working on a Java project where I would like to use Jade as the templating engine; however, what I really need is for the expression language to be Javascript based (i.e. using a custom Nashorn/Rhino context). We have a few reasons for this, but it mainly comes down to the API we are presenting for customisation is already in Javascript and we want to present a standard API to all parts of the application.

I have found other projects that have taken the Jade4J base to implement something similar (e.g. for JRuby https://github.com/guyboertje/jrjade); however, this question is whether the project would consider having the ability to provide alternate expression languages as a core feature (much like filters/plugins).

I have looked at how this could be implemented and it looks like it would be reasonably straight forward (especially for Javascript) and I would be happy to contribute, so I'm just looking for direction on whether this is something that would potentially be made part of the core project, or whether it would be a fork.

Thanks for your time.

P.S. Sorry for posting this as an issue. I looked for a mailing list but couldn't find one.

**Comment by chbloemer** [comment]
Thanks for your question.
With Jade 1.0.0 we added an ExpressionHandler interface so you can add your own implementation of an ExpressionHandler. Our intention was to try out a NashornExpressionHandler(see JsExpressionHandler.java) instead of a JexlExpressionHandler.

You can configure a different ExpressionHandler on the JadeConfiguration.
```java
JadeConfiguration jade = new JadeConfiguration();
jade.setExpressionHandler(new JsExpressionHandler());
JadeTemplate template = jade.getTemplate(file);
Writer writer = new StringWriter();
HashMap<String, Object> model = new HashMap<String, Object>();
jade.renderTemplate(template,model, writer);
String html = writer.toString();
```

Does this answer your question?
Hope I could help.

**Comment by cdjones32** [comment]
Hi Christoph,

Thankyou for the response, and yes that does answer the question. I had written and tested a stub Javascript ExpressionHandler myself (which is why I said it looked easy to do, because of the ExpressionHandler interface). I just missed that you actually already had the start of the Javascript one there already...

Awesome - I will start with that as a base and submit any pull requests against that.

Thanks again for the reply and apologies for missing that. I'm happy for this to be closed off as it isn't really in an issue format and I'll resubmit proper request when I submit a pull request.

**Comment by chbloemer** [comment]
Hi Chris,
how is your progress?

**Comment by cdjones32** [comment]
HI Christoph,

The project I was working on has been suspended for the moment, so I actually didn't get very far.  I will close this issue off as the original question was addressed and I am confident I would be able to do what I originally asked when I get back to it.

Thanks again for the response.

Regards
Chris
